test/pub.js

```diff
@@ -0,0 +1,57 @@
+const tape = require('tape')
+const crypto = require('crypto')
+const SecretStack = require('secret-stack')
+const { promisify: pify } = require('util')
+
+const u = require('./misc/util')
+
+const sleep = pify(setTimeout)
+const caps = {
+  shs: crypto.randomBytes(32).toString('base64'),
+}
+
+const CONNECTION_TIMEOUT = 500 // ms
+const REPLICATION_TIMEOUT = 2 * CONNECTION_TIMEOUT
+
+function Server(name, opts = {}) {
+  const stack = SecretStack({ caps }).use(require('ssb-db')).use(require('../'))
+
+  return stack({
+    temp: `test-pub-replicate-${name}`, // ssb-db only
+    timeout: CONNECTION_TIMEOUT,
+    keys: u.keysFor(name),
+    ...opts,
+  })
+}
+
+tape('Pub replicates Alice', async (t) => {
+  const aliceKeys = u.keysFor('alice')
+
+  const alice = Server('alice', { keys: aliceKeys })
+  const pub = Server('pub')
+
+  await pify(alice.publish)({ type: 'post', text: 'hello!' })
+  await pify(alice.publish)({ type: 'post', text: 'hello again' })
+
+  // Pub set up to want Alice's feed
+  pub.ebt.request(alice.id, true)
+  // alice.ebt.request(alice.id, true) // MYSTERY - replication fails without this
```

Comment:
# "**Solution**" :see_no_evil: 

if we have alice request herself before the connection, the replication to the pub works. This is very counter-intuitive. Seems like a bug not a feature.